orchestrator/gbt/src/client/deploy_erc20_representation.rs

```diff
@@ -99,9 +108,55 @@ pub async fn deploy_erc20_representation(
                     exit(0);
                 }
 
-                if Instant::now() - start > Duration::from_secs(100) {
-                    info!("Your ERC20 contract was not adopted, double check the metadata and try again");
-                    exit(1);
+                // we wait for up to WAIT_TIME seconds after that we must investigate why the attestation failed
+                const WAIT_TIME: u64 = 600;
+                if Instant::now() - start > Duration::from_secs(WAIT_TIME) {
+                    let attestations = grpc
+                        .get_attestations(QueryAttestationsRequest {
+                            limit: 0,
+                            order_by: String::new(),
+                            claim_type: String::new(),
+                            nonce: 0,
+                            height: 0,
+                        })
+                        .await;
+                    match attestations {
```

Comment:
It looks like this section wants to check the attestations returned every second, but if I understand the loop correctly, we will have the following:

1. Enter loop
2. Now - .0000001 seconds ago is not > 600 seconds
3. Wait for 1 second
Repeat 2 and 3 for 599 seconds
Finally check the adoption

Why not just delay for 600 seconds at the start?

Reply:
the logic here is not as robust as it should be. First off the diff is sort of hiding the exit condition, we can exit in less than 600 seconds during the happy path case where we find our new token in the erc20 mapping and exit successfully. 

We check for an update to the ERC20 mapping every second, but we only check the attestations at the end of the 600 second period. Ideally we would check both perhaps a little less frequently and error out if the attestation passes but the mapping does not update.